api/base/serializers.py

```diff
@@ -815,7 +835,7 @@ def to_representation(self, value):
                 return {'data': None}
 
         related_url = url['related']
-        related_path = urlparse(related_url).path
+        related_path = urlparse(related_url).path if related_url else None
```

Comment:
Allowing a self url and not a related url - for example, a preprint does not have a node relationship, but we still want to provide the self link for this.